### PR TITLE
Add condition for shallow clone

### LINKED ISSUE #52: [BUG] Integ test does a deep checkout of opensearch repository
**What is the bug?**
[Here](https://github.com/opensearch-project/opensearch-py/blob/340069f6fa344ab3853fec8c48e60e6a7756f271/test_opensearch/run_tests.py#L49), we're doing a deep checkout of OpenSearch (the server) repository. This works reasonably fast on Github Actions, but slows down the run time considerably when doing it on a developer laptop. We should ideally do a shallow checkout i.e. `git clone --depth=1 <url>`

**How can one reproduce the bug?**
Run `./.ci/run-test opensearch true` on dev laptop (requires Docker to be installed)

**What is the expected behavior?**
The build should finish fast

**What is your host/environment?**
 - OS: MacOS Catalina

**Do you have any screenshots?**
N/A

**Do you have any additional context?**
N/A

### PR DESCRIPTION
**Pull request by vchrombie**
### Description
The integration tests performs a deep checkout of the opensearch repository. Instead, we can perform a shallow clone the main branch to improve the speed and performance of the tests.
 
### Issues Resolved
Fixes #52

 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vchrombie** [comment]
Rebased the PR to resolve any merge conflicts.

**Comment by vchrombie** [comment]
@rushiagr do you mind reviewing the PR when you have time?

**Comment by vchrombie** [comment]
Hi @rushiagr, do you think you can review this PR when you have time?

**Review by axeoman — APPROVED** [review]
LGTM, thanks!
@VijayanB @deztructor any objects?

**Comment by vchrombie** [comment]
Thanks for the review @axeoman and sorry for the dismissal of the review. I rebased my branch to make sure all the checks are passing. Force push to the branch dismissed your review but I didn't intend to.

Thanks for understanding.

**Review by axeoman — APPROVED** [review]

**Comment by axeoman** [comment]
Sure no problem :+1: Thanks.
But let's wait for another approval anyway.

**Comment by axeoman** [comment]
Well, do not see any reasons not to improve CI that way and probably no harm if something else pops up anyway - so merging.

**Pull request merged**
